hoon/concepts/io.md

```diff
@@ -0,0 +1,471 @@
++++
```

Comment:
I'm actually thinking that this belongs in the `/docs/arvo` tree, since it's more about forms of IO throughout Urbit than it is about Hoon, seeing as it doesn't even use the Hoon language.